### DISCUSSION TITLE
[Feature] External tools

### DISCUSSION THREAD

**Discussion by aquaratixc** [FPGAwars/icestudio#840] [Ideas]
Please add External tools menu with ability to manage items in this menu i.e adding, remove and executing external commands

**Comment by cavearr** [comment]
Do you refer for example to open a shell console to execute commands?

**Comment by aquaratixc** [comment]
For example, I used icepll for calculation of frequencies for PLL and I
used own scripts written in Dlang for code generation

чт, 29 авг. 2019 г. в 23:42, Carlos Venegas Arrabé <notifications@github.com

**Comment by cavearr** [comment]
Ah! thanks for the explanation.
We are developing a plugin system to extend icestudio in this way, i'm update you by this issue.

**Comment by cavearr** [comment]
Hello! I am reorganizing the issues, trying to give more visibility and be able to better organize and prioritize the work and my roadmap (I will publish it very soon), I am converting some issues that are **new feature requests** as **ideas** on Github **discussions**.

Thanks for understanding.
